Core/Debouncer.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  Debouncer.swift
+//  DuckDuckGo
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+/// A class that provides a debouncing mechanism.
+public final class Debouncer {
```

Comment:
Seems old school to implement this with timers when Combine has one built in but:

1/ Then you'd need to pass in the debounce time in the constructor
2/ It's probably just as many lines of code as this

No change required.

Reply:
I’m fine to change to use Combine under the hood if we want, but yeah probably the lines of code will be very similar.